File: rating-service/src/main/java/com/astik/rating_service/controller/RatingController.java
package com.astik.rating_service.controller;

import com.astik.rating_service.dto.RatingRequest;
import com.astik.rating_service.dto.RatingResponse;
import com.astik.rating_service.service.RatingService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@Slf4j
@RequestMapping("/api/v1/ratings")
public class RatingController {
    private final RatingService ratingService;
    @PostMapping
    public ResponseEntity<RatingResponse>createRating(@Valid @RequestBody RatingRequest ratingRequest)
    {
        log.info("REST request to create Rating for user: {}", ratingRequest.getUserId());
        RatingResponse rating=ratingService.createRating(ratingRequest);
        return ResponseEntity.status(HttpStatus.CREATED).body(rating);
    }
    @GetMapping
    public ResponseEntity<List<RatingResponse>> getAllRatings() {
        log.info("REST request to get all Ratings");
        List<RatingResponse> ratings = ratingService.getAllRatings();
        return ResponseEntity.ok(ratings);
    }

    @GetMapping("/users/{userId}")
    public ResponseEntity<List<RatingResponse>> getRatingsByUserId(@PathVariable String userId) {
        log.info("REST request to get Ratings for user ID: {}", userId);
        List<RatingResponse> ratings = ratingService.getRatingByUserId(userId);
        return ResponseEntity.ok(ratings);
    }

    @GetMapping("/hotels/{hotelId}")
    public ResponseEntity<List<RatingResponse>> getRatingsByHotelId(@PathVariable String hotelId) {
        log.info("REST request to get Ratings for hotel ID: {}", hotelId);
        List<RatingResponse> ratings = ratingService.getRatingByHotelId(hotelId);
        return ResponseEntity.ok(ratings);
    }
}

File: pythonService/rating-service/controllers/rating_controller.py
from fastapi import APIRouter, status
from typing import List
import logging

from schemas.rating_request import RatingRequest
from schemas.rating_response import RatingResponse
from services.rating_service import RatingService

router = APIRouter(
    prefix="/api/v1/ratings",
    tags=["Ratings"]
)

logger = logging.getLogger(__name__)
rating_service = RatingService()


@router.post(
    "",
    response_model=RatingResponse,
    status_code=status.HTTP_201_CREATED
)
def create_rating(rating_request: RatingRequest):
    logger.info(f"REST request to create Rating for user: {rating_request.user_id}")
    return rating_service.create_rating(rating_request)


@router.get(
    "",
    response_model=List[RatingResponse]
)
def get_all_ratings():
    logger.info("REST request to get all Ratings")
    return rating_service.get_all_ratings()


@router.get(
    "/users/{user_id}",
    response_model=List[RatingResponse]
)
def get_ratings_by_user_id(user_id: str):
    logger.info(f"REST request to get Ratings for user ID: {user_id}")
    return rating_service.get_ratings_by_user_id(user_id)


@router.get(
    "/hotels/{hotel_id}",
    response_model=List[RatingResponse]
)
def get_ratings_by_hotel_id(hotel_id: str):
    logger.info(f"REST request to get Ratings for hotel ID: {hotel_id}")
    return rating_service.get_ratings_by_hotel_id(hotel_id)
